From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Fri, 14 Aug 2026 16:11:25 +0200
Subject: tests: Force C locale by unsetting LC_ALL

The power BatteryLevels tests match notification text against English
strings (e.g. "Battery Almost Empty"). To make this reliable the suite
sets LC_MESSAGES=C, but LC_ALL takes precedence over every other LC_*
variable. In environments where LC_ALL is set (e.g. LC_ALL=it_IT.UTF-8),
LC_MESSAGES=C is ignored and gsd-power emits translated notifications,
so the tests fail to find the expected strings.

Unset LC_ALL and LANGUAGE and force LANG=C.UTF-8 alongside LC_MESSAGES=C
so translations are actually disabled regardless of the surrounding
environment.

Origin: https://gitlab.gnome.org/GNOME/gnome-settings-daemon/-/merge_requests/489
---
 tests/gsdtestcase.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tests/gsdtestcase.py b/tests/gsdtestcase.py
index b58854d..ce75778 100644
--- a/tests/gsdtestcase.py
+++ b/tests/gsdtestcase.py
@@ -65,6 +65,9 @@ class GSDTestCase(DBusTestCase):
         os.environ['GIO_USE_VFS'] = 'local'
         os.environ['GVFS_DISABLE_FUSE'] = '1'
         # we do some string checks, disable translations
+        os.environ.pop('LC_ALL', None)
+        os.environ.pop('LANGUAGE', None)
+        os.environ['LANG'] = 'C.UTF-8'
         os.environ['LC_MESSAGES'] = 'C'
         klass.workdir = tempfile.mkdtemp(prefix='gsd-plugin-test')
         klass.addClassCleanup(shutil.rmtree, klass.workdir)
